### ISSUE TITLE
NS3 Make error with --enable-all-elements

### ISSUE THREAD

**Issue by lthurlow** [kohler/click#204]
```
./configure --disable-linuxmodule --enable-nsclick --enable-wifi --enable-all-elements
make
```

Does not build properly.

```
[ec2-user@ip-172-30-0-16 click]$ make
make[1]: Entering directory `/home/ec2-user/ns-allinone-3.23/click/userlevel'
  FINDELEMENTS elements.conf
  CREATE elements.mk
make[1]: Leaving directory `/home/ec2-user/ns-allinone-3.23/click/userlevel'
make[1]: Entering directory `/home/ec2-user/ns-allinone-3.23/click/userlevel'
  AR libclick.a
  RANLIB libclick.a
  CREATE elements.cc
  CXX elements.cc
  LINK click
ln: failed to create symbolic link ‘../bin/click’: File exists
make[1]: [click] Error 1 (ignored)
make[1]: Leaving directory `/home/ec2-user/ns-allinone-3.23/click/userlevel'
make[1]: Entering directory `/home/ec2-user/ns-allinone-3.23/click/ns'
echo "ns userlevel  analysis app aqm ethernet etherswitch grid icmp ip ip6 ipsec json local radio simple standard tcpudp test threads wifi" | ../bin/click-buildtool findelem -r ns -p .. -X ./elements.exclude --checksum elements.csmk > elements.conf
../bin/click-buildtool elem2make -x "addressinfo.o alignmentinfo.o errorelement.o portinfo.o scheduleinfo.o" < elements.conf > elements.mk
make[1]: Leaving directory `/home/ec2-user/ns-allinone-3.23/click/ns'
make[1]: Entering directory `/home/ec2-user/ns-allinone-3.23/click/ns'
  CXX ../elements/ipsec/hmacsha1.cc
In file included from ../elements/ipsec/hmacsha1.cc:112:0:
../elements/ipsec/hmac.cc: In function ‘void Click::HMAC_Init_ex(HMAC_CTX*, const void*, int)’:
../elements/ipsec/hmac.cc:80:26: error: call of overloaded ‘SHA1_init(SHA1_ctx*)’ is ambiguous
    SHA1_init(&ctx->md_ctx);
                          ^
../elements/ipsec/hmac.cc:80:26: note: candidates are:
In file included from ../elements/ipsec/hmacsha1.cc:111:0:
../elements/ipsec/sha1_impl.cc:285:1: note: void Click::SHA1_init(SHA1_ctx*)
 SHA1_init (SHA1_ctx * c)
 ^
In file included from ../elements/ipsec/hmac.hh:63:0,
                 from ../elements/ipsec/hmacsha1.cc:31:
../elements/ipsec/sha1_impl.hh:79:6: note: void SHA1_init(SHA1_ctx*)
 void SHA1_init (SHA1_ctx * c);
      ^
In file included from ../elements/ipsec/hmacsha1.cc:112:0:
../elements/ipsec/hmac.cc:81:52: error: call of overloaded ‘SHA1_update(SHA1_ctx*, unsigned char*, int&)’ is ambiguous
    SHA1_update(&ctx->md_ctx,(unsigned char*)key,len);
                                                    ^
../elements/ipsec/hmac.cc:81:52: note: candidates are:
In file included from ../elements/ipsec/hmacsha1.cc:111:0:
../elements/ipsec/sha1_impl.cc:298:1: note: void Click::SHA1_update(SHA1_ctx*, unsigned char*, long unsigned int)
 SHA1_update (SHA1_ctx * c, register unsigned char *data, unsigned long len)
 ^
In file included from ../elements/ipsec/hmac.hh:63:0,
                 from ../elements/ipsec/hmacsha1.cc:31:
../elements/ipsec/sha1_impl.hh:80:6: note: void SHA1_update(SHA1_ctx*, unsigned char*, long unsigned int)
 void SHA1_update (SHA1_ctx * c, unsigned char *data, unsigned long len);
      ^
In file included from ../elements/ipsec/hmacsha1.cc:112:0:
../elements/ipsec/hmac.cc:82:38: error: call of overloaded ‘SHA1_final(unsigned char [128], SHA1_ctx*)’ is ambiguous
    SHA1_final(ctx->key,&(ctx->md_ctx));
                                      ^
../elements/ipsec/hmac.cc:82:38: note: candidates are:
In file included from ../elements/ipsec/hmacsha1.cc:111:0:
../elements/ipsec/sha1_impl.cc:601:1: note: void Click::SHA1_final(unsigned char*, SHA1_ctx*)
 SHA1_final (unsigned char *md, SHA1_ctx *c)
 ^
In file included from ../elements/ipsec/hmac.hh:63:0,
                 from ../elements/ipsec/hmacsha1.cc:31:
../elements/ipsec/sha1_impl.hh:81:6: note: void SHA1_final(unsigned char*, SHA1_ctx*)
 void SHA1_final (unsigned char *md, SHA1_ctx * c);
      ^
In file included from ../elements/ipsec/hmacsha1.cc:112:0:
../elements/ipsec/hmac.cc:99:24: error: call of overloaded ‘SHA1_init(SHA1_ctx*)’ is ambiguous
   SHA1_init(&ctx->i_ctx);
                        ^
../elements/ipsec/hmac.cc:99:24: note: candidates are:
In file included from ../elements/ipsec/hmacsha1.cc:111:0:
../elements/ipsec/sha1_impl.cc:285:1: note: void Click::SHA1_init(SHA1_ctx*)
 SHA1_init (SHA1_ctx * c)
 ^
In file included from ../elements/ipsec/hmac.hh:63:0,
                 from ../elements/ipsec/hmacsha1.cc:31:
../elements/ipsec/sha1_impl.hh:79:6: note: void SHA1_init(SHA1_ctx*)
 void SHA1_init (SHA1_ctx * c);
      ^
In file included from ../elements/ipsec/hmacsha1.cc:112:0:
../elements/ipsec/hmac.cc:100:41: error: call of overloaded ‘SHA1_update(SHA1_ctx*, unsigned char [128], int)’ is ambiguous
   SHA1_update(&ctx->i_ctx,pad,SHA_CBLOCK);
                                         ^
../elements/ipsec/hmac.cc:100:41: note: candidates are:
In file included from ../elements/ipsec/hmacsha1.cc:111:0:
../elements/ipsec/sha1_impl.cc:298:1: note: void Click::SHA1_update(SHA1_ctx*, unsigned char*, long unsigned int)
 SHA1_update (SHA1_ctx * c, register unsigned char *data, unsigned long len)
 ^
In file included from ../elements/ipsec/hmac.hh:63:0,
                 from ../elements/ipsec/hmacsha1.cc:31:
../elements/ipsec/sha1_impl.hh:80:6: note: void SHA1_update(SHA1_ctx*, unsigned char*, long unsigned int)
 void SHA1_update (SHA1_ctx * c, unsigned char *data, unsigned long len);
      ^
In file included from ../elements/ipsec/hmacsha1.cc:112:0:
../elements/ipsec/hmac.cc:104:24: error: call of overloaded ‘SHA1_init(SHA1_ctx*)’ is ambiguous
   SHA1_init(&ctx->o_ctx);
                        ^
../elements/ipsec/hmac.cc:104:24: note: candidates are:
In file included from ../elements/ipsec/hmacsha1.cc:111:0:
../elements/ipsec/sha1_impl.cc:285:1: note: void Click::SHA1_init(SHA1_ctx*)
 SHA1_init (SHA1_ctx * c)
 ^
In file included from ../elements/ipsec/hmac.hh:63:0,
                 from ../elements/ipsec/hmacsha1.cc:31:
../elements/ipsec/sha1_impl.hh:79:6: note: void SHA1_init(SHA1_ctx*)
 void SHA1_init (SHA1_ctx * c);
      ^
In file included from ../elements/ipsec/hmacsha1.cc:112:0:
../elements/ipsec/hmac.cc:105:41: error: call of overloaded ‘SHA1_update(SHA1_ctx*, unsigned char [128], int)’ is ambiguous
   SHA1_update(&ctx->o_ctx,pad,SHA_CBLOCK);
                                         ^
../elements/ipsec/hmac.cc:105:41: note: candidates are:
In file included from ../elements/ipsec/hmacsha1.cc:111:0:
../elements/ipsec/sha1_impl.cc:298:1: note: void Click::SHA1_update(SHA1_ctx*, unsigned char*, long unsigned int)
 SHA1_update (SHA1_ctx * c, register unsigned char *data, unsigned long len)
 ^
In file included from ../elements/ipsec/hmac.hh:63:0,
                 from ../elements/ipsec/hmacsha1.cc:31:
../elements/ipsec/sha1_impl.hh:80:6: note: void SHA1_update(SHA1_ctx*, unsigned char*, long unsigned int)
 void SHA1_update (SHA1_ctx * c, unsigned char *data, unsigned long len);
      ^
In file included from ../elements/ipsec/hmacsha1.cc:112:0:
../elements/ipsec/hmac.cc: In function ‘void Click::HMAC_Init(HMAC_CTX*, const void*, int)’:
../elements/ipsec/hmac.cc:114:30: error: call of overloaded ‘HMAC_Init_ex(HMAC_CTX*&, const void*&, int&)’ is ambiguous
      HMAC_Init_ex(ctx,key,len);
                              ^
../elements/ipsec/hmac.cc:114:30: note: candidates are:
../elements/ipsec/hmac.cc:68:6: note: void Click::HMAC_Init_ex(HMAC_CTX*, const void*, int)
 void HMAC_Init_ex(HMAC_CTX *ctx, const void *key, int len)
      ^
In file included from ../elements/ipsec/hmacsha1.cc:31:0:
../elements/ipsec/hmac.hh:85:6: note: void HMAC_Init_ex(HMAC_CTX*, const void*, int)
 void HMAC_Init_ex(HMAC_CTX *ctx, const void *key, int len);
      ^
In file included from ../elements/ipsec/hmacsha1.cc:112:0:
../elements/ipsec/hmac.cc: In function ‘void Click::HMAC_Update(HMAC_CTX*, unsigned char*, size_t)’:
../elements/ipsec/hmac.cc:119:38: error: call of overloaded ‘SHA1_update(SHA1_ctx*, unsigned char*&, size_t&)’ is ambiguous
     SHA1_update(&ctx->md_ctx,data,len);
                                      ^
../elements/ipsec/hmac.cc:119:38: note: candidates are:
In file included from ../elements/ipsec/hmacsha1.cc:111:0:
../elements/ipsec/sha1_impl.cc:298:1: note: void Click::SHA1_update(SHA1_ctx*, unsigned char*, long unsigned int)
 SHA1_update (SHA1_ctx * c, register unsigned char *data, unsigned long len)
 ^
In file included from ../elements/ipsec/hmac.hh:63:0,
                 from ../elements/ipsec/hmacsha1.cc:31:
../elements/ipsec/sha1_impl.hh:80:6: note: void SHA1_update(SHA1_ctx*, unsigned char*, long unsigned int)
 void SHA1_update (SHA1_ctx * c, unsigned char *data, unsigned long len);
      ^
In file included from ../elements/ipsec/hmacsha1.cc:112:0:
../elements/ipsec/hmac.cc: In function ‘void Click::HMAC_Final(HMAC_CTX*, unsigned char*, unsigned int*)’:
../elements/ipsec/hmac.cc:128:30: error: call of overloaded ‘SHA1_final(unsigned char [64], SHA1_ctx*)’ is ambiguous
   SHA1_final(buf,&ctx->md_ctx);
                              ^
../elements/ipsec/hmac.cc:128:30: note: candidates are:
In file included from ../elements/ipsec/hmacsha1.cc:111:0:
../elements/ipsec/sha1_impl.cc:601:1: note: void Click::SHA1_final(unsigned char*, SHA1_ctx*)
 SHA1_final (unsigned char *md, SHA1_ctx *c)
 ^
In file included from ../elements/ipsec/hmac.hh:63:0,
                 from ../elements/ipsec/hmacsha1.cc:31:
../elements/ipsec/sha1_impl.hh:81:6: note: void SHA1_final(unsigned char*, SHA1_ctx*)
 void SHA1_final (unsigned char *md, SHA1_ctx * c);
      ^
In file included from ../elements/ipsec/hmacsha1.cc:112:0:
../elements/ipsec/hmac.cc:130:51: error: call of overloaded ‘SHA1_update(SHA1_ctx*, unsigned char [64], long unsigned int)’ is ambiguous
   SHA1_update(&ctx->md_ctx,buf,(unsigned long)*len);
                                                   ^
../elements/ipsec/hmac.cc:130:51: note: candidates are:
In file included from ../elements/ipsec/hmacsha1.cc:111:0:
../elements/ipsec/sha1_impl.cc:298:1: note: void Click::SHA1_update(SHA1_ctx*, unsigned char*, long unsigned int)
 SHA1_update (SHA1_ctx * c, register unsigned char *data, unsigned long len)
 ^
In file included from ../elements/ipsec/hmac.hh:63:0,
                 from ../elements/ipsec/hmacsha1.cc:31:
../elements/ipsec/sha1_impl.hh:80:6: note: void SHA1_update(SHA1_ctx*, unsigned char*, long unsigned int)
 void SHA1_update (SHA1_ctx * c, unsigned char *data, unsigned long len);
      ^
In file included from ../elements/ipsec/hmacsha1.cc:112:0:
../elements/ipsec/hmac.cc:131:29: error: call of overloaded ‘SHA1_final(unsigned char*&, SHA1_ctx*)’ is ambiguous
   SHA1_final(md,&ctx->md_ctx);
                             ^
../elements/ipsec/hmac.cc:131:29: note: candidates are:
In file included from ../elements/ipsec/hmacsha1.cc:111:0:
../elements/ipsec/sha1_impl.cc:601:1: note: void Click::SHA1_final(unsigned char*, SHA1_ctx*)
 SHA1_final (unsigned char *md, SHA1_ctx *c)
 ^
In file included from ../elements/ipsec/hmac.hh:63:0,
                 from ../elements/ipsec/hmacsha1.cc:31:
../elements/ipsec/sha1_impl.hh:81:6: note: void SHA1_final(unsigned char*, SHA1_ctx*)
 void SHA1_final (unsigned char *md, SHA1_ctx * c);
      ^
In file included from ../elements/ipsec/hmacsha1.cc:112:0:
../elements/ipsec/hmac.cc: In function ‘void Click::HMAC_CTX_init(HMAC_CTX*)’:
../elements/ipsec/hmac.cc:136:24: error: call of overloaded ‘SHA1_init(SHA1_ctx*)’ is ambiguous
   SHA1_init(&ctx->i_ctx);
                        ^
../elements/ipsec/hmac.cc:136:24: note: candidates are:
In file included from ../elements/ipsec/hmacsha1.cc:111:0:
../elements/ipsec/sha1_impl.cc:285:1: note: void Click::SHA1_init(SHA1_ctx*)
 SHA1_init (SHA1_ctx * c)
 ^
In file included from ../elements/ipsec/hmac.hh:63:0,
                 from ../elements/ipsec/hmacsha1.cc:31:
../elements/ipsec/sha1_impl.hh:79:6: note: void SHA1_init(SHA1_ctx*)
 void SHA1_init (SHA1_ctx * c);
      ^
In file included from ../elements/ipsec/hmacsha1.cc:112:0:
../elements/ipsec/hmac.cc:137:24: error: call of overloaded ‘SHA1_init(SHA1_ctx*)’ is ambiguous
   SHA1_init(&ctx->o_ctx);
                        ^
../elements/ipsec/hmac.cc:137:24: note: candidates are:
In file included from ../elements/ipsec/hmacsha1.cc:111:0:
../elements/ipsec/sha1_impl.cc:285:1: note: void Click::SHA1_init(SHA1_ctx*)
 SHA1_init (SHA1_ctx * c)
 ^
In file included from ../elements/ipsec/hmac.hh:63:0,
                 from ../elements/ipsec/hmacsha1.cc:31:
../elements/ipsec/sha1_impl.hh:79:6: note: void SHA1_init(SHA1_ctx*)
 void SHA1_init (SHA1_ctx * c);
      ^
In file included from ../elements/ipsec/hmacsha1.cc:112:0:
../elements/ipsec/hmac.cc:138:25: error: call of overloaded ‘SHA1_init(SHA1_ctx*)’ is ambiguous
   SHA1_init(&ctx->md_ctx);
                         ^
../elements/ipsec/hmac.cc:138:25: note: candidates are:
In file included from ../elements/ipsec/hmacsha1.cc:111:0:
../elements/ipsec/sha1_impl.cc:285:1: note: void Click::SHA1_init(SHA1_ctx*)
 SHA1_init (SHA1_ctx * c)
 ^
In file included from ../elements/ipsec/hmac.hh:63:0,
                 from ../elements/ipsec/hmacsha1.cc:31:
../elements/ipsec/sha1_impl.hh:79:6: note: void SHA1_init(SHA1_ctx*)
 void SHA1_init (SHA1_ctx * c);
      ^
In file included from ../elements/ipsec/hmacsha1.cc:112:0:
../elements/ipsec/hmac.cc: In function ‘unsigned char* Click::HMAC(void*, int, unsigned char*, size_t, unsigned char*, unsigned int*)’:
../elements/ipsec/hmac.cc:152:19: error: call of overloaded ‘HMAC_CTX_init(HMAC_CTX*)’ is ambiguous
   HMAC_CTX_init(&c);
                   ^
../elements/ipsec/hmac.cc:152:19: note: candidates are:
../elements/ipsec/hmac.cc:134:6: note: void Click::HMAC_CTX_init(HMAC_CTX*)
 void HMAC_CTX_init(HMAC_CTX *ctx)
      ^
In file included from ../elements/ipsec/hmacsha1.cc:31:0:
../elements/ipsec/hmac.hh:81:6: note: void HMAC_CTX_init(HMAC_CTX*)
 void HMAC_CTX_init(HMAC_CTX *ctx);
      ^
In file included from ../elements/ipsec/hmacsha1.cc:112:0:
../elements/ipsec/hmac.cc:153:27: error: call of overloaded ‘HMAC_Init(HMAC_CTX*, void*&, int&)’ is ambiguous
   HMAC_Init(&c,key,key_len);
                           ^
../elements/ipsec/hmac.cc:153:27: note: candidates are:
../elements/ipsec/hmac.cc:110:6: note: void Click::HMAC_Init(HMAC_CTX*, const void*, int)
 void HMAC_Init(HMAC_CTX *ctx, const void *key, int len)
      ^
In file included from ../elements/ipsec/hmacsha1.cc:31:0:
../elements/ipsec/hmac.hh:84:6: note: void HMAC_Init(HMAC_CTX*, const void*, int)
 void HMAC_Init(HMAC_CTX *ctx, const void *key, int len); /* deprecated */
      ^
In file included from ../elements/ipsec/hmacsha1.cc:112:0:
../elements/ipsec/hmac.cc:154:21: error: call of overloaded ‘HMAC_Update(HMAC_CTX*, unsigned char*&, size_t&)’ is ambiguous
   HMAC_Update(&c,d,n);
                     ^
../elements/ipsec/hmac.cc:154:21: note: candidates are:
../elements/ipsec/hmac.cc:117:6: note: void Click::HMAC_Update(HMAC_CTX*, unsigned char*, size_t)
 void HMAC_Update(HMAC_CTX *ctx, unsigned char *data, size_t len)
      ^
In file included from ../elements/ipsec/hmacsha1.cc:31:0:
../elements/ipsec/hmac.hh:86:6: note: void HMAC_Update(HMAC_CTX*, unsigned char*, size_t)
 void HMAC_Update(HMAC_CTX *ctx, unsigned char *data, size_t len);
      ^
In file included from ../elements/ipsec/hmacsha1.cc:112:0:
../elements/ipsec/hmac.cc:155:26: error: call of overloaded ‘HMAC_Final(HMAC_CTX*, unsigned char*&, unsigned int*&’ is ambiguous
   HMAC_Final(&c,md,md_len);
                          ^
../elements/ipsec/hmac.cc:155:26: note: candidates are:
../elements/ipsec/hmac.cc:122:6: note: void Click::HMAC_Final(HMAC_CTX*, unsigned char*, unsigned int*)
 void HMAC_Final(HMAC_CTX *ctx, unsigned char *md, unsigned int *len)
      ^
In file included from ../elements/ipsec/hmacsha1.cc:31:0:
../elements/ipsec/hmac.hh:87:6: note: void HMAC_Final(HMAC_CTX*, unsigned char*, unsigned int*)
 void HMAC_Final(HMAC_CTX *ctx, unsigned char * md, unsigned int *len);
      ^
In file included from ../elements/ipsec/hmacsha1.cc:112:0:
../elements/ipsec/hmac.cc:156:22: error: call of overloaded ‘HMAC_CTX_cleanup(HMAC_CTX*)’ is ambiguous
   HMAC_CTX_cleanup(&c);
                      ^
../elements/ipsec/hmac.cc:156:22: note: candidates are:
../elements/ipsec/hmac.cc:141:6: note: void Click::HMAC_CTX_cleanup(HMAC_CTX*)
 void HMAC_CTX_cleanup(HMAC_CTX *ctx)
      ^
In file included from ../elements/ipsec/hmacsha1.cc:31:0:
../elements/ipsec/hmac.hh:82:6: note: void HMAC_CTX_cleanup(HMAC_CTX*)
 void HMAC_CTX_cleanup(HMAC_CTX *ctx);
      ^
make[1]: *** [hmacsha1.o] Error 1
make[1]: Leaving directory `/home/ec2-user/ns-allinone-3.23/click/ns'
make: *** [ns] Error 2
```

**Comment by lthurlow** [comment]
Only happen when --enable-wifi and --enable-all-elements are both set.  bad configure command.

**Comment by kohler** [comment]
Actually, this failure was because the IPsec elements were incompatible with the ns driver. I believe that issue has been fixed by 9e0e09509345090d432916fbeb608c9610424a11
